### PR TITLE
Work with records

### PR DESCRIPTION
**Pull request by sanderson**
- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/flux/v0.x/data-types/composite/record.md [hunk 1]
```diff
@@ -10,4 +10,168 @@ menu:
     parent: Composite types
 weight: 201
 flux/v0.x/tags: ["composite types", "data types"]
----
\ No newline at end of file
+---
+
+A **record** type is a set of key-value pairs (also known as properties).
```
Comment (kelseiv) [inline content/flux/v0.x/data-types/composite/record.md:15]: Should we define properties in our glossary (are we using this term throughout the docs in the same way?). 
Noticed our definition of field refers to "the key-value pair in InfluxDB’s data structure that records metadata and the actual data value." 

Would it be accurate to say a record is a set of fields (or tags?) also known as properties? If not, maybe helpful to clarify this distinction as appropriate throughout.

File: content/flux/v0.x/data-types/composite/record.md [hunk 2]
```diff
@@ -10,4 +10,168 @@ menu:
...
+- [Record syntax](#record-syntax)
+- [Reference values in a record](#reference-values-in-a-record)
+- [Operate on records](#operate-on-records)
+
+## Record syntax
+A **record** literal contains a set of key-value pairs (properties) enclosed in curly brackets (`{}`).
+Properties are comma-delimitted.
+**Property keys must be strings** and can optionally be enclosed in double quotes (`"`).
+If a property key contains only numeric characters or any whitespace characters,
+it must be enclosed in double quotes.
```
Comment (kelseiv) [inline content/flux/v0.x/data-types/composite/record.md:30]:
```suggestion
If a property key contains only numeric characters and whitespace characters,
you must enclose the property key in double quotes.
```

File: content/flux/v0.x/data-types/composite/record.md [hunk 3]
```diff
@@ -10,4 +10,168 @@ menu:
...
+  "Company Name": "ACME",
+  "Street Address": "123 Main St.",
+  id: 1123445
+}
+
+c["Company Name"]
+// Returns ACME
+
+c["id"]
+// Returns 1123445
```
Comment (kelseiv) [inline content/flux/v0.x/data-types/composite/record.md:87]: did you mean to use Street Address as and example....or was this intentional to show you can get the id with bracket or dot notation?

**Review by kelseiv — APPROVED** [review]

File: content/flux/v0.x/data-types/composite/record.md (hunk 1 above)
Comment (sanderson) [reply content/flux/v0.x/data-types/composite/record.md:15]: No, the term "properties" is specific to records. The terms "fields" or "tags" don't really have meaning within the Flux language. They are concepts tied to the InfluxDB storage engine.

File: content/flux/v0.x/data-types/composite/record.md (hunk 3 above)
Comment (sanderson) [reply content/flux/v0.x/data-types/composite/record.md:87]: Both :). It was to show how to format a key that has a space it it. You have to wrap it in quotes.

**Pull request merged**